tests/drivers/input/kbd_matrix/src/main.c

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/input/input_kbd_matrix.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#define TEST_KBD_SCAN_NODE DT_INST(0, test_kbd_scan)
+
+/* test driver */
+
+static struct {
+	kbd_row_t rows[3];
```

Comment:
Also add a comment: "Provide mock row data for every valid column".

Reply:
Done.